packages/components/src/components/table/table-head-cell/table-head-cell-styles.ts

```diff
@@ -38,28 +46,53 @@ export const getComponentCss = (
       ...(sortable
         ? {
             button: {
+              transform: 'translate3d(0,0,0)', // creates new stacking context
               display: 'flex',
-              alignItems: 'flex-end',
+              gap: spacingStaticXSmall,
+              width: 'auto',
+              margin: 0,
               padding: 0,
-              boxSizing: 'border-box',
-              appearance: 'none',
-              border: 'none',
+              color: 'inherit',
+              outline: 0,
               ...textSmallStyle,
-              fontWeight: fontWeightSemiBold,
-              color: primaryColor,
-              textDecoration: 'none',
-              textAlign: 'left',
+              justifyContent: 'flex-start',
+              alignItems: 'flex-end',
+              fontSize: getFontSizeText('small'),
+              appearance: 'none',
               background: 'transparent',
+              textAlign: 'left',
+              border: 0,
               cursor: 'pointer',
-              ...getFocusJssStyle({ offset: 1 }),
+              '&::before': {
+                content: '""',
+                position: 'absolute', // mobile Safari -> prevent lagging active state
+                top: offsetVertical,
+                bottom: offsetVertical,
+                right: offsetHorizontal,
+                left: offsetHorizontal,
+                borderRadius: borderRadiusSmall,
+                transition: getTransition('background-color'),
+              },
               ...hoverMediaQuery({
-                ...getHoverJssStyle(),
                 '&:hover, &:focus': {
                   '& .icon': {
                     opacity: 1,
                   },
                 },
+                '&:hover::before': {
+                  ...frostedGlassStyle,
+                  backgroundColor: hoverColor,
+                },
               }),
+              '&:focus::before': {
+                border: `${borderWidthBase} solid ${focusColor}`,
+              },
+              '&:not(:focus-visible)::before': {
+                border: 0,
+              },
+            },
+            '.label': {
```

Comment:
`z-index: -1` on the hover pseudo and `z-index: 0` on the `button` would achieve the same without the additional node.

Reply:
Fixed it